backend/src/field_filter.rs

```diff
@@ -34,10 +34,10 @@ macro_rules! generalize_factory {
                             unsafe {
                                 let result = <$restricted_factory as BackendFactory<$supported_type>>::create(
                                     &$restricted_factory,
-                                    std::mem::transmute(pil), std::mem::transmute(fixed), output_dir, setup,
+                                    std::mem::transmute::<std::sync::Arc<powdr_ast::analyzed::Analyzed<F>>, std::sync::Arc<powdr_ast::analyzed::Analyzed<$supported_type>>>(pil), std::mem::transmute::<std::sync::Arc<std::vec::Vec<(std::string::String, powdr_executor::constant_evaluator::VariablySizedColumn<F>)>>, std::sync::Arc<std::vec::Vec<(std::string::String, powdr_executor::constant_evaluator::VariablySizedColumn<$supported_type>)>>>(fixed), output_dir, setup,
```

Comment:
1) what

Reply:
Wait this evil macro was merged after all?

Reply:
> Wait this evil macro was merged after all?

The first `unsafe` solution got nuked by rust. The evil macro was the only PR in town.
https://github.com/rust-lang/rust/pull/120248#issuecomment-2340261183